redbot/cogs/warnings/warnings.py

```diff
@@ -392,27 +392,45 @@ async def warn(
             member = user
         elif isinstance(user, int):
             if not ctx.channel.permissions_for(ctx.guild.me).ban_members:
-                await ctx.send(_("User {user} is not in the server."),format(user=user))
+                await ctx.send(_("I cannot ban users. Missing the permission to ban members."))
                 return
             user_obj = self.bot.get_user(user) or discord.Object(id=user)
-            try:
-                confirm = ConfirmView(ctx.author, timeout=30)
-                confirm.message = await ctx.send(
-                    _("User `{user}` is not in the server. Would you like to ban them instead?").format(user=user),
-                    view=confirm,
-                )
-                await confirm.wait()
-                if confirm.result:
+
+            confirm = ConfirmView(ctx.author, timeout=30)
+            confirm.message = await ctx.send(
+                _(
+                    "User `{user}` is not in the server. Would you like to ban them instead?"
+                ).format(user=user),
+                view=confirm,
+            )
+            await confirm.wait()
+            if confirm.result:
+                try:
                     await ctx.guild.ban(user_obj, reason=reason)
-                else:
-                    confirm.message = await ctx.send(_("No action taken."))
+                    await modlog.create_case(
+                        self.bot,
+                        guild,
+                        ctx.message.created_at,
+                        "ban",
+                        user,
+                        ctx.author,
+                        reason,
+                        until=None,
+                        channel=None,
+                    )
+                except discord.HTTPException as error:
+                    await ctx.send(
+                        _("An error occurred while trying to ban the user. Error: {error}").format(
+                            error=error
+                        )
+                    )
+            else:
+                confirm.message = await ctx.send(_("No action taken."))
 
+            if ctx.channel.permissions_for(ctx.guild.me).add_reactions:
                 await confirm.message.add_reaction("\N{WHITE HEAVY CHECK MARK}")
-
-            except discord.Forbidden:
-                await ctx.send(_("I don't have permission to ban this user."))
-            except discord.HTTPException:
-                await ctx.send(_("An error occurred while trying to ban the user."))
+            else:
+                await ctx.tick()
```

Comment:
`ctx.tick` is a shorthand for `if permissions_for.add_reactions: add_reaction(CHECK)`. That's why I said before you could just replace the `add_reaction` call with a call to `tick`, and let its error handling do the trick.

Reply:
There is always the potential for error, but if an error you cannot think of happens, you probably _don't_ want to suppress it. You want to be able to identify it so you know if it is something that needs to be dealt with. Any API call can error if the internet goes out, but it doesn't make sense to wrap every API call in a `try` just in case.

I don't think it hurts to keep an `HTTPException` `try` on the `ban` call, since there are other cases that do make sense to catch, such as if the user is already banned. But the bot not having permissions is an obvious one that makes sense to have additional pre-check for. The `try` on the `tick` would be unnecessary.

Reply:
I'm not sure which ones you mean, in this file at least they are all using `tick`.

Reply:
nevermind. I swore I saw there was some mention of it. Must've been in a different file.